app/src/main/java/org/alphatilesapps/alphatiles/Validator.java

```diff
@@ -0,0 +1,992 @@
+            super(driveFolderId, inName, "application/vnd.google-apps.folder");
+            String pageToken = null;
+            do {
+                FileList result = driveService.files().list()
+                        .setQ("parents in '" + driveFolderId + "' and trashed = false")
+                        .setSpaces("drive")
+                        .setFields("nextPageToken, files(id, name, mimeType)")
+                        .setPageToken(pageToken)
+                        .execute();
+
+                for (File file : result.getFiles()){
+
+                    if (file.getMimeType().equals("application/vnd.google-apps.spreadsheet")) {
+                        folderContents.add(new GoogleSheet(file.getId(), file.getName()));
+                    }
+                    else if (file.getMimeType().equals("application/vnd.google-apps.folder")) {
+                        folderContents.add(new DriveFolder(file.getId(), file.getName()));
+                    }
+                    else {
+                        folderContents.add(new GoogleDriveItem(file.getId(), file.getName(), file.getMimeType()));
+                    }
+                 }
+                pageToken = result.getNextPageToken();
+            } while (pageToken != null);
+        }
+
+        protected DriveFolder getFolderFromName(String inName) throws ValidatorException{
+            for (DriveFolder item : this.getContentsOfType(DriveFolder.class)){
+                if (item.getName().equals(inName)){
+                    return item;
+                }
+            }
+            throw new ValidatorException("was not able to find " + inName + " in the drive folder " + this.getName());
+        }
+
+        protected void checkFileNameAgainstList(ArrayList<String> toMatch){
+            PriorityQueue<String> LongestFirstToMatch = new PriorityQueue<>(toMatch);
+            for (GoogleDriveItem item : new ArrayList<>(folderContents)) {
+                boolean hasMatchingName = false;
+                for (String candidate : new ArrayList<>(LongestFirstToMatch)) {
+                    if (item.getName().startsWith(candidate)) {
+                        LongestFirstToMatch.remove(candidate);
+                        hasMatchingName = true;
+                    }
+                }
+                if (!hasMatchingName) {
+                    warnings.add("the file " + item.getName() + " in " + this.getName() + " may be excess" +
+                            "as the start of the filename does not appear to match to anything");
+                    folderContents.remove(item);
+                }
+            }
+            for (String shouldHaveMatched : LongestFirstToMatch){
+                warnings.add(shouldHaveMatched + "does not have a corresponding file in " + this.getName() +
+                        " of the correct file type");
+            }
+            //TODO should I have services.json?
```

Comment:
I think adding google-services.json is a separate task from validating and building the rest of the language pack. It is a one-time file creation through Firebase that just gets inserted into the product flavor folder independent of other files.